File: datacapturing/src/main/java/de/cyface/datacapturing/exception/SetupException.java
package de.cyface.datacapturing.exception;

import androidx.annotation.NonNull;

/**
 * An <code>Exception</code> that is thrown each time setting up the <code>DataCapturingService</code> fails.
 *
 * @author Klemens Muthmann
 * @version 1.0.0
 * @since 2.0.0
 */
public final class SetupException extends Exception {
    /**
     * Creates a new completely initialized <code>SetupException</code>, wrapping another <code>Exception</code> from
     * deeper within the system.
     *
     * @param e The wrapped <code>Exception</code>.
     */
    public SetupException(final @NonNull Exception e) {
        super(e);
    }

    /**
     * Creates a new completely initialized <code>SetupException</code>, providing a detailed explanation about the
     * error.
     *
     * @param message The message explaining the error condition.
     */
    public SetupException(final @NonNull String message) {
        super(message);
    }
}

File: persistence/src/main/kotlin/de/cyface/persistence/SetupException.kt
/*
 * Copyright 2023 Cyface GmbH
 *
 * This file is part of the Cyface SDK for Android.
 *
 * The Cyface SDK for Android is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The Cyface SDK for Android is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with the Cyface SDK for Android. If not, see <http://www.gnu.org/licenses/>.
 */
package de.cyface.persistence

/**
 * An `Exception` that is thrown each time setting up the `DataCapturingService` fails.
 *
 * @author Klemens Muthmann
 * @version 1.0.2
 * @since 2.0.0
 */
class SetupException : Exception {
    /**
     * Creates a new completely initialized `SetupException`, wrapping another `Exception` from
     * deeper within the system.
     *
     * @param e The wrapped `Exception`.
     */
    constructor(e: Exception) : super(e)

    /**
     * Creates a new completely initialized `SetupException`, providing a detailed explanation about the
     * error.
     *
     * @param message The message explaining the error condition.
     */
    constructor(message: String) : super(message)
}